nixos/modules/system/boot/plymouth.nix

```diff
@@ -19,30 +19,10 @@ let
     osVersion = config.system.nixos.release;
   };
 
-  plymouthLogos = pkgs.runCommand "plymouth-logos" { inherit (cfg) logo; } ''
-    mkdir -p $out
-
-    # For themes that are compiled with PLYMOUTH_LOGO_FILE
-    mkdir -p $out/etc/plymouth
-    ln -s $logo $out/etc/plymouth/logo.png
-
-    # Logo for bgrt theme
-    # Note this is technically an abuse of watermark for the bgrt theme
-    # See: https://gitlab.freedesktop.org/plymouth/plymouth/-/issues/95#note_813768
-    mkdir -p $out/share/plymouth/themes/spinner
-    ln -s $logo $out/share/plymouth/themes/spinner/watermark.png
-
-    # Logo for spinfinity theme
-    # See: https://gitlab.freedesktop.org/plymouth/plymouth/-/issues/106
-    mkdir -p $out/share/plymouth/themes/spinfinity
-    ln -s $logo $out/share/plymouth/themes/spinfinity/header-image.png
-  '';
-
```

Comment:
Is this not sufficient for stage-1?

https://github.com/NixOS/nixpkgs/blob/54b1d155046952c41c79bb0a7da834f52effe3b4/nixos/modules/system/boot/plymouth.nix#L164

Or if it is maybe keep the first five lines and have the rest of the symlinks in the derivation?

Reply:
No because the extra-utils build logic tries to resolve symlinks when copying for reasons that aren't clear to me. But also having "installed" files in the nix store for an average package that point out of the Nix store to files that possibly do not exist and are system-config dependent feels wrong to me (but it does feel fine for, e.g., NixOS modules to do this since they define /etc anyway)